pkg/discovery/discovery.go

```diff
@@ -219,3 +229,87 @@ func (d *tidbDiscovery) DiscoverDM(advertisePeerUrl string) (string, error) {
 	delete(currentCluster.peers, podName)
 	return fmt.Sprintf("--join=%s", strings.Join(mastersArr, ",")), nil
 }
+
+func (d *tidbDiscovery) VerifyPDEndpoint(advertisePeerURL string) (string, error) {
```

Comment:
exported function should have a brief comment on its main logic, such as
```
//VerifyPDEndpoint takes in an pd peer url, and checks whether it is available. If yes, ... , if no, ... .
```

Please also add comments in the function body, for any nontrivial code block.